src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -342,5 +352,36 @@ private static string GetNupkgUrl(string id, string version)
 
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE0044:Add readonly modifier", Justification = "Modified through reflection.")]
         private static string ENV_NUGET_HOMEPAGE = "https://www.nuget.org/packages";
+        
+        
+        public record NuGetArtifactUri : BaseArtifactUri
+        {
+            /// <summary>
+            /// Initializes a new instance of <see cref="NuGetArtifactUri"/>.
+            /// </summary>
+            /// <param name="type">The type of artifact for this <see cref="NuGetArtifactUri"/>.</param>
+            /// <param name="uri">The <see cref="Uri"/> this artifact can be found at.</param>
+            /// <param name="extension">The file extension for the file found at the <paramref name="uri"/>.</param>
+            public NuGetArtifactUri(NuGetArtifactType type, Uri uri, string? extension = null) : 
+                base(type, uri, extension ?? Path.GetExtension(uri.AbsolutePath))
+            {
+            }
+
+            /// <summary>
+            /// Initializes a new instance of <see cref="NuGetArtifactUri"/>.
+            /// </summary>
+            /// <param name="type">The type of artifact for this <see cref="NuGetArtifactUri"/>.</param>
+            /// <param name="uri">The string of the uri this artifact can be found at.</param>
+            /// <param name="extension">The file extension for the file found at the <paramref name="uri"/>.</param>
+            public NuGetArtifactUri(NuGetArtifactType type, string uri, string? extension = null) : this(type, new Uri(uri), extension) { }
+        }
+
+        public enum NuGetArtifactType
+        {
+            Unknown = 0,
+            Binary,
+            Nupkg,
+            Nuspec,
```

Comment:
Are you going to add these to the enumeration above in this PR?

Reply:
I removed `Binary` but I am doing `Nuspec` now too.